packages/backend-for-frontend/src/services/toolService.ts

```diff
@@ -41,31 +41,43 @@ import {
   multipleAgreementForEserviceAndConsumer,
   organizationNotAllowed,
   purposeIdNotFoundInClientAssertion,
+  invalidClientAssertionType,
+  invalidGrantType,
 } from "../model/errors.js";
-import { TokenGenerationValidationStepFailure } from "../../../api-clients/dist/bffApi.js";
 import {
   PagoPAInteropBeClients,
   AgreementProcessClient,
   CatalogProcessClient,
 } from "../clients/clientsProvider.js";
 import { getAllAgreements, getLatestAgreement } from "./agreementService.js";
 
+const DEFAULT_CLIENT_ASSERTION_TYPE =
+  "urn:ietf:params:oauth:client-assertion-type:jwt-bearer";
+const DEFAULT_GRANT_TYPE = "client_credentials";
+
 export function toolsServiceBuilder(clients: PagoPAInteropBeClients) {
   return {
     async validateTokenGeneration(
       clientId: string | undefined,
       clientAssertion: string,
-      _clientAssertionType: string,
-      _grantType: string,
+      clientAssertionType: string,
+      grantType: string,
       ctx: WithLogger<BffAppContext>
     ): Promise<bffApi.TokenGenerationValidationResult> {
       ctx.logger.info(`Validating token generation for client ${clientId}`);
 
+      if (clientAssertionType !== DEFAULT_CLIENT_ASSERTION_TYPE) {
+        throw invalidClientAssertionType();
+      }
+
+      if (grantType !== DEFAULT_GRANT_TYPE) {
+        throw invalidGrantType();
+      }
```

Comment:
I think we don't even need this validation here: the client assertion validation already does this check.
See: https://vscode.dev/github/pagopa/interop-be-monorepo/blob/main/packages/client-assertion-validation/src/validation.ts#L53-L54

Let's leave the validation logic on the client assertion validation side, and here we just pass the parameters to it as they come. In this way, the client assertion validation can support a new values in the future and this code would still work without changes

Reply:
We can ask this to @galales or @taglioni-r , maybe we should change the accepted types with a string and not a strict const of the types.